### ISSUE TITLE
yarn install报错：gifsicle pre-build test failed

### ISSUE THREAD

**Issue by xietongjian** [vbenjs/vue-vben-admin#1438]
有没有人遇到以下这个问题？
vben-2.6.1版本
在运行yarn install的时候一直报错，clean:lib，clean:cache都没用，删除-lock文件也没用，卡了好久了

`
warning " > vite-plugin-pwa@0.8.2" has unmet peer dependency "workbox-window@^6.1.5".
[5/6] Building fresh packages...
[8/13] ⠈ optipng-bin
[6/13] ⠈ jpegtran-bin
[9/13] ⠈ pngquant-bin
[7/13] ⠈ mozjpeg
error D:\workspace\xxx\xxx\node_modules\vite-plugin-imagemin\node_modules\gifsicle:
Command failed.
Exit code: 1
Command: node lib/install.js
Arguments:
Directory: D:\workspace\xxx\xxx\node_modules\vite-plugin-imagemin\node_modules\gifsi
cle
Output:
Response code 404 (Not Found)
gifsicle pre-build test failed
compiling from source
Error: Command failed: C:\WINDOWS\system32\cmd.exe /s /c "autoreconf -ivf"
'autoreconf' �����ڲ����ⲿ���Ҳ���ǿ����еĳ���
���������ļ���

    at D:\workspace\xxx\xxx\node_modules\bin-build\node_modules\execa\index.js:231:1
1

**Comment by Rorschach331** [comment]
我也是，
win11
node:v16.13.1
yarn:1.22.10

```
Output:
Response code 404 (Not Found)
gifsicle pre-build test failed
compiling from source
Error: Command failed: C:\WINDOWS\system32\cmd.exe /s /c "autoreconf -ivf"
'autoreconf' �����ڲ����ⲿ���Ҳ���ǿ����еĳ���
���������ļ���
```

**Comment by xietongjian** [comment]
> 我也是， win11 node:v16.13.1 yarn:1.22.10
> 
> ```
> Output:
> Response code 404 (Not Found)
> gifsicle pre-build test failed
> compiling from source
> Error: Command failed: C:\WINDOWS\system32\cmd.exe /s /c "autoreconf -ivf"
> 'autoreconf' �����ڲ����ⲿ���Ҳ���ǿ����еĳ���
> ���������ļ���
> ```
我找到原因了，还原这个文件就可以了yarn.lock。这个文件不能删除。

**Comment by Rorschach331** [comment]
我换成pnpm，把官方的pnpm-lock.yaml放进项目里，好了

**Comment by Tanimodori** [comment]
**TL;DR:** Change the `resolutions` section in `package.json` as below:

```json
  "resolutions": {
    "bin-wrapper": "npm:bin-wrapper-china@0.1.0",
    "gifsicle": "5.2.0",
    "rollup": "^2.56.3"
  },
```

This fix also applies to [anncwb/vben-admin-thin-next](https://github.com/anncwb/vben-admin-thin-next)

The package `bin-wrapper-china` has the `gifsicle-bin` download issue by applying mirrors in `binary-mirror-config`, but the windows executable for `v5.3.0` is missing in `npmmirror.com` (i.e. `GET https://npmmirror.com/mirrors/gifsicle-bin/v5.3.0/vendor/win/x64/gifsicle.exe 404`). Since both `v5.2.0` and `v6.1.0` exist and similar fixes (anncwb/vue-vben-admin#1497, anncwb/vue-vben-admin@2aabbc788ab05d06f4dcc2a01b25585ecf8f690b) uses `v5.2.0`, this fix chose that version as the resolution.

The `@0.1.0` suffix is required in Yarn 2 protocal schemas (see https://yarnpkg.com/features/protocols). This fix is tested to successfully installed with Yarn `v1.22.11` /`v3.1.1` and Node `v16.13.1` . Supposely this fix and those applied at [anncwb/vue-vben-admin](https://github.com/anncwb/vue-vben-admin) can fix anncwb/vue-vben-admin#345 and anncwb/vue-vben-admin#1438

Note for Yarn proxy settings: 

* For Yarn 1, use `yarn config set proxy http://host:port` and `yarn config set https-proxy http://host:port` in terminal, or you can edit `~/.yarnrc` directly.
* For Yarn 2 and up, use `yarn config set httpProxy http://host:port` and `yarn config set httpsProxy http://host:port` in terminal, or you can edit `~/.yarnrc.yml` directly. Note that `~/.yarnrc` will **not** work with Yarn 2.
